### PR TITLE
Added CLI command for stopping a deployment

### PR DESCRIPTION
**Pull request by tpoitras**
Docs trello card: https://trello.com/c/uh3DegFd
Dev trello card: https://trello.com/c/QLdFlyA4

Waiting on dev work to complete but getting a jumpstart on documenting this option to stop a deployment.

@ironcladlou -- I know you're still working on this but I thought I'd get going. Take a look and let me know if I'm missing anything. Can you maybe give me a use case or two for WHY a user would want to stop a deployment?

### CODE REVIEW

File: cli_reference/basic_cli_operations.adoc [hunk 1]
```diff
@@ -353,3 +353,14 @@ To perform the rollback manually by piping the *JSON* of the new configuration b
 $ osc rollback deployment-1 --output=json | osc update deploymentConfigs deployment -f -
 ----
 ====
+
+*Stopping a Deployment*
```
Comment (abhgupta) [inline cli_reference/basic_cli_operations.adoc]: The term we use is "Cancel".

**Comment by abhgupta** [comment]
There could be several use cases for cancelling a deployment:
1. A deployment could be running for too long and perhaps is stuck. With a deployment already running, another deployment cannot be initiated. Cancellation can especially come in handy when a custom deployment strategy (deployer pod implementation) is used.
2. A deployment was initiated incorrectly and the user wishes to take corrective action.
3. A security fix is required to be applied and the user does not want to wait for an ongoing deployment to complete.

**Comment by abhgupta** [comment]
The implementation PR has been merged: https://github.com/openshift/origin/pull/2468

**Comment by abhgupta** [comment]
@tpoitras Did you have any more questions related to this feature?

**Comment by tpoitras** [comment]
@abhgupta -- why is cancellation useful when using a custom deployment strategy (deployer pod 
implementation)?

**Comment by abhgupta** [comment]
@tpoitras Cancellation is always useful. We will test and harden our own deployment strategies to ensure that they do not get "stuck" or create issues. With custom deployment strategies, cancellation gives the user an additional safety-net to control/manage misbehaving deployments. Similarly can come in handy in cases when hook implementations have bugs and need to be stopped immediately.

Come to think of it, I would not make a distinction between our out-of-the-box deployment strategies vs custom strategies. I would just simply restate point 1 as cancellation being useful if the deployment logic or hooks are "stuck" and running for too long.

**Comment by abhgupta** [comment]
@timothypoitras

Also, I would like a mention in the docs of the fact that by default, the deployment process is restricted to 6 hours. If the deployment is still in-progress after 6 hours, the containers for the deployer pod and any deployment hook pods are deleted and these pods are placed in the failed state.

This duration is not user changeable at this time.

**Comment by tpoitras** [comment]
Thanks, @abhgupta :)
Please check this over and let me know if it's all good now. :+1:

**Comment by abhgupta** [comment]
LGTM

**Comment by adellape** [comment]
Peer reviewed, LGTM. Merging.

**Pull request merged**